### PR TITLE
Fetch run trace and output in parallel

### PR DESCRIPTION
**Pull request by amh4r**
## Description
Fetch run trace and output in parallel.

This required fetching the run output from a resolver that results a string instead of parsed output. That's a change we needed anyway since the output parsing in the backend has some edge case bugs

## Type of change (choose one)
- [ ] Chore (refactors, upgrades, etc.)
- [x] Bug fix (non-breaking change that fixes an issue)
- [ ] Security fix (non-breaking change that fixes a potential vulnerability)
- [ ] Docs
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality not to work as expected)

## Checklist
- [ ] I've linked any associated issues to this PR.
- [ ] I've tested my own changes.

*[Check our Pull Request Guidelines](https://github.com/inngest/inngest/blob/main/docs/PULL_REQUEST_GUIDELINES.md)*

### CODE REVIEW

File: ui/packages/components/src/RunResult.tsx [hunk 1]
```diff
@@ -1,46 +1,91 @@
...
       )}
 
-      {result.error && (
+      {!parsedResult.isSuccess && (
         <CodeBlock
           header={{
-            title:
-              (result.error.name || 'Error') +
-              (result.error.message ? ': ' + result.error.message : ''),
+            title: `${parsedResult.error.name}: ${parsedResult.error.message}`,
```
Comment (djfarrelly) [inline ui/packages/components/src/RunResult.tsx:34]: **issue**: Only add the colon when there is a message like we did before
```suggestion
            title: `${parsedResult.error.name}${parsedResult.error.message?.length > ": ": ""}${parsedResult.error.message}`,
```

File: ui/packages/components/src/RunResult.tsx [hunk 2]
```diff
@@ -1,46 +1,91 @@
...
   className?: string;
-  result: Result;
-  isSuccess?: boolean;
+  result: string;
+  isSuccess: boolean;
 };
 
 export function RunResult({ className, result, isSuccess }: Props) {
-  const prettyResult = result.data && usePrettyJson(result.data);
+  const parsedResult = useParsedResult({ isSuccess, result });
```
Comment (djfarrelly) [inline ui/packages/components/src/RunResult.tsx:15]: **question**: Before this, we used `result.data` and now `useParsedResult` is using `result` - is this due to the way that the different API endpoint returns the response? Or is the string `result` include the `data` key? 

I ask this because I'm looking for it to be the same experience, and not have the successful output nested within `{ "data": { "message": "cool" }` - Confirming it should show to the user as `{ "message": "cool" }.

File: ui/apps/dashboard/src/components/RunDetails/useGetTraceResult.ts [hunk 3]
```diff
@@ -1,35 +1,29 @@
...
+  query TraceResult($envID: ID!, $runID: String!) {
     workspace(id: $envID) {
-      runTraceSpanOutputByID(outputID: $traceID) {
-        data
-        error {
-          message
-          name
-          stack
-        }
+      run(runID: $runID) {
```
Comment (djfarrelly) [inline ui/apps/dashboard/src/components/RunDetails/useGetTraceResult.ts:10]: **question**: does this endpoint also use Clickhouse as the source of truth or is this an old endpoint that uses Big Query? I want to make sure that it uses the same source of truth as the rest of the new UI

File: ui/packages/components/src/RunDetailsV2/RunDetails.tsx [hunk 4]
```diff
@@ -57,13 +56,8 @@ export function RunDetails(props: Props) {
     enabled: Boolean(outputID),
```
Comment (djfarrelly) [inline ui/packages/components/src/RunDetailsV2/RunDetails.tsx:56]: **issue**: It still looks like this query will not execute until the `outputID` is fetched so this won't run in parallel. I think this needs to change to `runID` or remove the enabled config option all together as everything else is passed by props.

**Review by djfarrelly — CHANGES_REQUESTED** [review]
Hey @goodoldneon, I have some questions and have highlighted a couple of issues that I think might block this from actually running in parallel. Catch me on Slack if you want to discuss live!

File: ui/packages/components/src/RunResult.tsx (hunk 2 above)
Comment (goodoldneon) [reply ui/packages/components/src/RunResult.tsx:15]: We had to make this change because the normal run resolver returns unstructured output data. The old, non-parallelizable one returned structured data. However, there are edge cases the structured resolver didn't handle.

I don't see nested data when I test it:
image

File: ui/apps/dashboard/src/components/RunDetails/useGetTraceResult.ts (hunk 3 above)
Comment (goodoldneon) [reply ui/apps/dashboard/src/components/RunDetails/useGetTraceResult.ts:10]: It's ClickHouse. This is the same field that the run details queries:
https://github.com/inngest/inngest/blob/aaron/inn-3422-execute-run-trace-and-output-queries-in-parallel-not/ui/apps/dashboard/src/components/RunDetails/useGetRun.ts#L46

File: ui/packages/components/src/RunDetailsV2/RunDetails.tsx (hunk 4 above)
Comment (goodoldneon) [reply ui/packages/components/src/RunDetailsV2/RunDetails.tsx:56]: Oops, good call

File: ui/packages/components/src/RunResult.tsx (hunk 2 above)
Comment (djfarrelly) [reply ui/packages/components/src/RunResult.tsx:15]: Ok great! Just wanted to confirm the output wasn't nested as we've had regressions in the past.
